app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3667,41 +3661,44 @@ private void setupSubtitleView() {
         binding.subtitleView.setStyle(captionStyle);
     }
 
-    private void onTextTracksChanged() {
-        final int textRenderer = getCaptionRendererIndex();
-
+    private void onTextTracksChanged(@NonNull final TracksInfo currentTrackInfo) {
         if (binding == null) {
             return;
         }
-        if (trackSelector.getCurrentMappedTrackInfo() == null
-                || textRenderer == RENDERER_UNAVAILABLE) {
+
+        if (!currentTrackInfo.isTypeSupportedOrEmpty(C.TRACK_TYPE_TEXT)) {
```

Comment:
when there is no subtitle track available, this method would return true. I think you need to restore the previous conditions